dockerfile/parser/testfiles/heredoc/Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM docker.io/library/python:latest
+RUN <<EOF
+echo "Hello" >> /hello
+echo "World!" >> /hello
+EOF
+
+
+RUN python3 <<EOF
+with open("/hello", "w") as f:
+    print("Hello", file=f)
+    print("Something", file=f)
+EOF
+
+ADD <<EOF /index.html
+(your index page goes here)
+EOF
+
+COPY <<robots.txt <<humans.txt /test/
+(robots content)
+robots.txt
+(humans content)
+humans.txt
+
+RUN cat index.html
+RUN cat hello
```

Comment:
I'd like to see some test coverage of the `FD<<word`, `<<-`, and `<<"word"` variations, both by themselves and in combinations.

Reply:
Done, for othercases but i think `<<"word"` is not a valid case ( it does not works with buildkit as well I have verified manually ). For all othercases extended the test.

Reply:
My bad I misunderstood with terminator also having quotes i.e `"EOF"`. The dockerfile which you have pasted is valid and I have added a case for the Dockerfile which you have provided.

The bad dockerfile which i was trying
```Dockerfile
# syntax=docker/dockerfile:1.4
FROM alpine
RUN <<"EOF"
id > /tmp/id.txt
"EOF"
```

Reply:
We still need at least one test where the descriptor number isn't 0, something like
```Dockerfile
RUN 5<<EOF cat /proc/self/fd/5 > file.txt
this is the file
EOF
```

Reply:
Done.

Reply:
Done added few more cases which should cover this.

Reply:
All of the tests in here include a blank line after the terminator, so we're not going to notice if we start skipping the next line after the terminator when it isn't blank.  Please add a case where the final heredoc terminator for an instruction line is immediately followed by a non-empty line.

Reply:
done.